src/library_pthread.js

```diff
@@ -321,7 +321,12 @@ var LibraryPThread = {
         // Ask the new worker to load up the Emscripten-compiled page. This is a heavy operation.
         worker.postMessage({
             cmd: 'load',
-            url: currentScriptUrl,
+            // If the application main .js file was loaded from a Blob, then it is not possible
+            // to access the URL of the current script that could be passed to a Web Worker so that
+            // it could load up the same file. In that case, developer must either deliver the Blob
```

Comment:
could we always create a blob here and send that over?

Reply:
Technically we could, but doing that has disadvantages, the URL changes from human readable one to a blob UUID one, which affects CSP and CORS behavior, profiling, printed callstacks and logged messages, and we'd have to add a new extra code to do the XHRs in. It would be preferred to allow users to keep loading the `.js` files when they want to do that.

Reply:
Got it, lgtm.